charts/argo-cd/templates/argocd-application-controller/deployment.yaml

```diff
@@ -57,9 +57,9 @@ spec:
         - --loglevel
         - {{ .Values.controller.logLevel }}
         {{- if .Values.controller.metrics.applicationLabels.enabled }}
-        - --metrics-application-labels
         {{- range .Values.controller.metrics.applicationLabels.labels }}
-        - {{- toYaml . | nindent 8 }}
+        - --metrics-application-labels
+        - {{ . }}
         {{- end }}
         {{- end }}
```

Comment:
> @mkilchhofer can you clarify what change you are suggesting. Are you saying you would still like the toYaml so it can take a list of maps? e.g. `{{ . }}` -> `{{ toYaml . }}` Does it make sense to do this?
> 
> ```
> labels:
>  - { foo: bar }
> ```



No, you moved the `--metrics-application-labels` into the range block. Like this every item in the list of strings will be "prefixed" with `--metrics-application-labels` which I think is not necessary.

Reply:
I was following the docs here -> https://argo-cd.readthedocs.io/en/stable/operator-manual/metrics/#exposing-application-labels-as-prometheus-metrics

But if you say it is not necessary, I can change

Reply:
Moved `--metrics-application-labels` outside of range block as suggested